etl/steps/data/meadow/hmd/2022-11-03/hmd_lt.py

```diff
@@ -0,0 +1,322 @@
+    # Set dtypes
+    df = _clean_set_dtypes_df(df)
+    return df
+
+
+def _clean_rename_columns_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Rename columns."""
+    return df.rename(
+        columns={
+            "Country": "country",
+            "Year": "year",
+            "Age": "age",
+            "mx": "central_death_rate",
+            "qx": "probability_of_death",
+            "ax": "avg_survival_length",
+            "lx": "num_survivors",
+            "dx": "num_deaths",
+            "Lx": "num_person_years_lived",
+            "Tx": "num_person_years_remaining",
+            "ex": "life_expectancy",
+        }
+    )
+
+
+def _clean_correct_missing_data(df: pd.DataFrame) -> pd.DataFrame:
+    """Checks on missing data."""
+    # Expected missing data
+    num_rows_missing_expected = 0.01
+    countries_missing_data_expected = 1
+    # Find missing data
+    rows_missing = df[df.central_death_rate == "."]
+    num_rows_missing = len(rows_missing) / len(df)
+    countries_missing_data = rows_missing.country.unique()
+    # Run checks
+    assert num_rows_missing < num_rows_missing_expected, (
+        f"More missing data than expected was found! {round(num_rows_missing*100, 2)} rows missing, but"
+        f" {round(num_rows_missing_expected*100,2)} were expected."
+    )
+    assert len(countries_missing_data) <= countries_missing_data_expected, (
+        f"More missing data than expected was found! Found {len(countries_missing_data)} countries, expected is"
+        f" {countries_missing_data_expected}. Check {countries_missing_data}!"
+    )
+    # Correct
+    df = df.replace(".", pd.NA)
+    return df
+
+
+def _clean_set_dtypes_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Set dtypes."""
+    # dtypes
+    dtypes = {
+        "central_death_rate": "Float64",
+        "probability_of_death": "Float64",
+        "avg_survival_length": "Float64",
+        "num_survivors": "Int64",
+        "num_deaths": "Int64",
+        "num_person_years_lived": "Int64",
+        "num_person_years_remaining": "Int64",
+        "life_expectancy": "Float64",
+        "age": "category",
+        "country": "category",
+        "year": "category",
+    }
+    # Numeric
+    cols_numeric = [col_name for col_name, dtype in dtypes.items() if dtype in ["Int64", "Float64"]]
+    for col in cols_numeric:
+        df[col] = pd.to_numeric(df[col])
+    return df.astype(dtypes)
+
+
+def df_to_table(walden_ds: WaldenCatalog, table_name: str, df: pd.DataFrame) -> catalog.Table:
+    """Convert plain pandas.DataFrame into table.
+
+    Parameters
+    ----------
+    walden_ds : WaldenCatalog
+        Raw Walden dataset.
+    table_name : str
+        Table name.
+    df : pd.DataFrame
+        Dataframe.
+
+    Returns
+    -------
+    catalog.Table
+        Table created from dataframe, walden metadata and table name.
+    """
+    # create table with metadata from dataframe
+    table_metadata = TableMeta(
+        short_name=f"agexyear_{table_name}",
```

Comment:
Maybe think of a different name for the tables?

Reply:
I improved the description of the table so the user can have a better understanding of what the table contains.